common/config/persistence.go

```diff
@@ -41,6 +41,11 @@ const (
 	StoreTypeNoSQL = "nosql"
 )
 
+var (
+	ErrInvalidVisibilityStoreCombination = errors.New("persistence config: Cannot set visibilityStore and secondaryVisibilityStore with different datastore types")
+	ErrConfigMismatch                    = errors.New("persistence config: Config mismatch for visibilityStore and secondaryVisibilityStore")
+)
```

Comment:
Create a base error, and then use below like this:
```
var ErrPersistenceConfig = errors.New("persistence config error")


return fmt.Errorf("%w: config mismatch for visibilityStore and secondaryVisibilityStore", ErrPersistenceConfig)
```

Reply:
I created a base error to be "persistence config" as we are wrapping the validation error already. This is what error looks like: `config validation error: persistence config: error messages`